packages/dart_frog_cli/e2e/test/new_test.dart

```diff
@@ -0,0 +1,484 @@
+
+      await dartFrogNewRoute(
+        '/some_other_route/deep/deep/internal',
+        directory: projectDirectory,
+      );
+
+      expect(
+        fileAt('some_other_route.dart', on: routesDirectory),
+        doesNotExist,
+      );
+      expect(
+        fileAt('some_other_route/index.dart', on: routesDirectory),
+        exists,
+      );
+      expect(
+        fileAt(
+          'some_other_route/deep/deep/internal.dart',
+          on: routesDirectory,
+        ),
+        exists,
+      );
+    });
+
+    test(
+      'Creates route normally when there is a non-dart file with the same '
+      'route path',
+      () async {
+        File(path.join(routesDirectory.path, 'something.py'))
+            .createSync(recursive: true);
+
+        await dartFrogNewRoute('/something', directory: projectDirectory);
+
+        expect(fileAt('something.dart', on: routesDirectory), exists);
+        expect(fileAt('something.py', on: routesDirectory), exists);
+      },
+    );
+
+    test('Excuse root route', () async {
+      await expectLater(
+        () async => dartFrogNewRoute('/', directory: projectDirectory),
+        failsWithA(message: 'Failed to create route: / already exists.'),
+      );
+    });
+
+    test('Excuse existing endpoints', () async {
+      await dartFrogNewRoute('/existing_endpoint', directory: projectDirectory);
+
+      await expectLater(
+        () async =>
+            dartFrogNewRoute('/existing_endpoint', directory: projectDirectory),
+        failsWithA(
+          message: 'Failed to create route: /existing_endpoint already exists.',
+        ),
+      );
+    });
+
+    test('Excuse existing endpoints (indexed route)', () async {
+      await dartFrogNewRoute(
+        '/existing_endpoint_dir',
+        directory: projectDirectory,
+      );
+
+      await dartFrogNewRoute(
+        '/existing_endpoint_dir/inside',
+        directory: projectDirectory,
+      );
+
+      await expectLater(
+        () async => dartFrogNewRoute(
+          '/existing_endpoint_dir',
+          directory: projectDirectory,
+        ),
+        failsWithA(
+          message:
+              'Failed to create route: /existing_endpoint_dir already exists.',
+        ),
+      );
+    });
+
+    test('Excuse route creation of invalid route identifiers', () async {
+      await expectLater(
+        () async => dartFrogNewRoute('/👯‍', directory: projectDirectory),
+        failsWithA(message: 'Route path must be a valid path segment.\n\n'),
+      );
+    });
+
+    test('Excuse route creation of doubled route params', () async {
+      await expectLater(
+        () async => dartFrogNewRoute(
+          '/[id]/something/[id]',
+          directory: projectDirectory,
+        ),
+        failsWithA(
+          message: 'Failed to create route: Duplicate parameter name found: id',
+        ),
+      );
+    });
+
+    group('Invalid states', () {
+      // These tests create invalid states that may break other tests if
```

Comment:
If this is the case can we split this each e2e each into a separate file? I think @wolfenrain had a reason behind splitting e2e tests into one per file as in [Very Good CLI](https://github.com/VeryGoodOpenSource/very_good_cli/tree/main/e2e/test/commands/test).